sympy/ntheory/factor_.py

```diff
@@ -1419,6 +1419,82 @@ def divisor_count(n, modulus=1):
     return Mul(*[v + 1 for k, v in factorint(n).items() if k > 1])
 
 
+def proper_divisors(n, generator=False):
```

Comment:
Let's give `_divisors` a 2nd argument, `proper=False` and call `_divisors` from `proper_divisors` rather than duplicating code:
```python
def _divisors(n, proper=False):
    ...
    if not proper or p != n: yield p
```

Reply:
How about this?
```python
if proper:
    for p in rec_gen():
        if p != n:
            yield p
else:
    for p in rec_gen():
        yield p
```

Reply:
And a generator argument as well? Seemed like a neat idea (but I haven't checked the code so I cannot tell how feasible it is).

Reply:
I think I didn't notice the _...

Reply:
@smichr Added ```proper``` argument. Thanks!